src/modules/job-manager/update.c

```diff
@@ -70,50 +68,67 @@
 struct update {
     struct job_manager *ctx;
     flux_msg_handler_t **handlers;
+    zlistx_t *pending_requests;
+};
+
+struct update_request {
+    void *handle;
+    struct update *update;
+    const flux_msg_t *msg;
+    struct flux_msg_cred cred;
+    struct job *job;
+    json_t *updates;
+    unsigned int validate:1;
```

Comment:
An inline comment that `handle` is a zlistx handle for this item in `update->pending_requests` might be helpful.